expression/builtin_string.go

```diff
@@ -2701,10 +2701,31 @@ func (b *builtinOrdSig) evalInt(row chunk.Row) (int64, bool, error) {
 	if isNull || err != nil {
 		return 0, isNull, err
 	}
-	return ord(str), false, nil
+
+	charSet := b.args[0].GetType().Charset
+	// use utf8 by default
+	if charSet == "" {
+		charSet = charset.CharsetUTF8
+	}
+	desc, err := charset.GetCharsetDesc(charSet)
+	if err != nil {
+		return 0, isNull, err
+	}
+
+	if desc.Maxlen == 1 {
+		return ordBinary(str), false, nil
+	}
+	return ordUtf8(str), false, nil
+}
```

Comment:
Please extract these code as a function name by `chooseOrdFunc` or other. Then we can reuse the function in vecEvalInt.

Reply:
Finished